src/operator/controllers/intents_reconcilers/networkpolicy/reconciler.go

```diff
@@ -309,28 +326,68 @@ func (r *Reconciler) buildNetworkPolicy(ctx context.Context, ep effectivepolicy.
 		return nil, false, nil
 	}
 
-	policy := v1.NetworkPolicy{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf(otterizev2alpha1.OtterizeSingleNetworkPolicyNameTemplate, ep.Service.GetNameWithKind()),
-			Namespace: ep.Service.Namespace,
-			Labels: map[string]string{
-				otterizev2alpha1.OtterizeNetworkPolicy: ep.Service.GetFormattedOtterizeIdentityWithKind(),
+	if r.ShouldCreateSeparatePolicies {
+		if shouldCreateIngress {
```

Comment:
The nesting level here is tough. 
consider splitting this to `createSeparatePolicies` and `createSinglePolicy` functions, and then this would change to:
```
if r.ShouldCreateSeparatePolicies {
    networkPolicies = r.createSeparatePolicies(...)
} else {
    ... r.createSinglePolicy(...)
}
...
```